sympy/physics/mechanics/abstract_body.py

```diff
@@ -0,0 +1,136 @@
+from abc import ABC, abstractmethod
+
+from sympy.core.backend import Symbol, sympify
+from sympy.physics.vector import Point
+
+__all__ = ['_Body']
+
+
+class _Body(ABC):
+    """Abstract class for body type objects."""
+    def __init__(self, name, masscenter=None, mass=None):
+        if not isinstance(name, str):
+            raise TypeError('Supply a valid name.')
+        self._name = name
+        if mass is None:
+            mass = Symbol(f'{name}_mass')
+        if masscenter is None:
+            masscenter = Point(f'{name}_masscenter')
+        self.mass = mass
+        self.masscenter = masscenter
+        self.potential_energy = 0
+        self.points = []
+
+    def __str__(self):
+        return self.name
+
+    def __repr__(self):
+        return self.__str__()
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    @abstractmethod
+    def frame(self):
+        """The ReferenceFrame fixed to the body."""
+        pass
+
+    @frame.setter
+    @abstractmethod
+    def frame(self, frame):
+        pass
```

Comment:
Docstring?

Reply:
Should the `setter` also have a docstring, if the `property` already has one? This is most of the time not done in SymPy.

Reply:
No. This was my mistake. I didn't realise that this was the setter.